### PR TITLE
SPR-4691 - Add SimpleJdbcUpdate

### PR DESCRIPTION
**Pull request by benelog**
I have signed and agree to the terms of the Spring Individual Contributor License Agreement.

This patch is about https://jira.spring.io/browse/SPR-4691

### CODE REVIEW

File: spring-jdbc/src/main/java/org/springframework/jdbc/core/simple/AbstractJdbcUpdate.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	/**
+	 * Constructor for sublasses to delegate to for setting the JdbcTemplate.
+	 */
+	protected AbstractJdbcUpdate(JdbcTemplate jdbcTemplate) {
+		super(jdbcTemplate);
+	}
+
+
+	// -------------------------------------------------------------------------
+	// Methods dealing with configuaration properties
```
Comment (tomsquest) [inline spring-jdbc/src/main/java/org/springframework/jdbc/core/simple/AbstractJdbcUpdate.java]: typo `configuaration` -> `configuration`

File: spring-jdbc/src/main/java/org/springframework/jdbc/core/simple/AbstractJdbcUpdate.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+	/**
+	 * Constructor for sublasses to delegate to for setting the JdbcTemplate.
+	 */
+	protected AbstractJdbcUpdate(JdbcTemplate jdbcTemplate) {
+		super(jdbcTemplate);
+	}
+
+
+	// -------------------------------------------------------------------------
+	// Methods dealing with configuration properties
```
Comment (benelog) [inline spring-jdbc/src/main/java/org/springframework/jdbc/core/simple/AbstractJdbcUpdate.java]: @tomsquest
I fixed typo ( configuaration -> configuration ), and rebased on master again.

**Comment by benelog** [comment]
@snicoll 
Is there any feedback on this?

**Comment by snicoll** [comment]
Sorry, I hadn't a chance to look at this yet. It's on my todo list.

**Comment by kervin521** [comment]
I can find or query Entity(s),Only to set Property of Entity,but auto update to data base,I can't saveOrUpdate operation.

for example:

List<A> list = commonService.findByCriteria(criteria);
for(A a:list){
    a.setStatus(1);// default is 0;
}
follow up,A-->status=1 will auto update to data base;
why?
how to handle to avoid to happen?

protect frame:
1.spring 4.3.2+hibernate 5.1.0
2.spring 3.2.17+hibernate 3.6.10
The above two combinations have existed this problem

**Comment by snicoll** [comment]
@kervin521 I am not sure I follow and this PR is not a place for questions. Please raise them on Stackoverflow. Thanks!

**Comment by danesavot** [comment]
Any update about this PR approval? SimpleJdbcUpdate would come in handy.

**Comment by benelog** [comment]
@snicoll 
This PR widely affects AbstractJdbcInsert, TableMetaDataContext. As I mentioned on the JIRA issue, this task may be suitable for maintainers of Spring framework.

https://jira.spring.io/browse/SPR-4691?focusedCommentId=128931&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-128931

I resolved conflicts, but feel free to close the PR. I expect that the Spring framework team will solve it in a better way.

**Comment by benelog** [comment]
Since Spring Data JDBC is considered a better alternative to SimpleJdbcUpdate, I will close this PR.

**Comment by sasynkamil** [comment]
> Since Spring Data JDBC is considered a better alternative to SimpleJdbcUpdate, I will close this PR.

Isn't '[Spring Data JDBC](https://docs.spring.io/spring-data/jdbc/docs/2.1.6/reference/html/#jdbc.why)' higher level of abstraction than '[Spring JDBC](https://docs.spring.io/spring-framework/docs/current/reference/html/data-access.html#jdbc)'?

**Comment by benelog** [comment]
@sasynkamil 
You're right.
My comment was about my experiences.
After using Spring Data JDBC, I don't need the SimpleJdbcUpdate any more, so I closed this PR.

**Pull request closed without merging**